searchcore/src/vespa/searchcore/proton/server/proton.cpp

```diff
@@ -674,7 +674,7 @@ Proton::ping(MonitorRequest::UP request, MonitorClient & client)
     ret.partid = protonConfig.partition;
     ret.distribution_key = protonConfig.distributionkey;
     ret.timestamp = (_matchEngine->isOnline()) ? 42 : 0;
-    ret.activeDocs = getNumActiveDocs();
+    ret.activeDocs = (_matchEngine->isOnline()) ? getNumActiveDocs() : 0;
```

Comment:
This guard would not be needed if the new search interface remained down until transaction log replay had completed, just like the old search interface.

Reply:
I did not find any use of the ability to not listen:

[0] 2918aaac101d:~/git/vespa$ git grep "setListen("
searchcore/src/vespa/searchcore/fdispatch/program/fdispatch.cpp:            _transportServer->setListen(false);
searchcore/src/vespa/searchcore/fdispatch/program/fdispatch.cpp:            _transportServer->setListen(true);
searchlib/src/vespa/searchlib/engine/transportserver.h:    void setListen(bool listen) {